app/views/posts/_post_subject.html.erb

```diff
@@ -1,14 +1,17 @@
 <% post = post_subject %>
 <% cache_unless(post.recent?, post, namespace: 'post_subject', expires_in: 72.hours, race_condition_ttl: 30.seconds) do %>
-<span class="short-post-subject"><%= link_to raw(htmlic_title(post.subject)), post %></span> by
-<span class="short-post-author author-tag-for-<%= post.user_id %>"><%= post.author %></span>
-<span class="short-post-time">on <%= post.created_at %></span>
-<%= raw '<span class="badge" label="New post" title="New post">&nbsp;</span>' if post.new_reply? %>
-<%= raw '<span class="label label-warning">Locked</span>' if post.locked %>
-<%= raw '<span class="label label-danger">Poofed</span>' if post.poofed %>
-<%= raw '<span class="label label-info">Edited</span>' if post.previous_version %>
-<%= raw '<span class="label label-important">Previous Version</span>' if post.next_version %>
-<% post.tags.each do |tag| %>
-<span class="label label-default short-post-tag"><%= tag.name %></span>
-<% end %>
+  <span class="short-post-subject"><%= link_to raw(htmlic_title(post.subject)), post %></span> by
+  <span class="short-post-author author-tag-for-<%= post.user_id %>"><%= post.author %></span>
+  <span class="short-post-time">on <%= post.created_at %></span>
+  <%= raw '<span class="badge" label="New post" title="New post">&nbsp;</span>' if post.new_reply? %>
+  <%= raw '<span class="label label-warning">Locked</span>' if post.locked %>
+  <%= raw '<span class="label label-danger">Poofed</span>' if post.poofed %>
+  <%= raw '<span class="label label-info">Edited</span>' if post.previous_version %>
+  <%= raw '<span class="label label-important">Previous Version</span>' if post.next_version %>
+  <% unless post.has_parent? %>
+    <span class="label label-primary"><%= post.descendant_ids.length %> replies</span>
```

Comment:
I wonder what query that issues and whether it's unnecessarily materializing the list. (aka, I'm curious if this just ends up being a `SELECT COUNT(*)`, and this should be tellable from the logs)

Reply:
As far as I can tell, it's using existing data. When you pull a post from the DB currently it also pulls the replies nestled under it, as in the main view, so it seems to be pulling all the necessary from the info when you initially load the page.

The README for the Ancestry library notes that you can pull all ActiveRecords in a single query (https://github.com/stefankroes/ancestry), implying this is the case. You get a Post object from the DB, you also fetch all its ancestors, meaning no additional queries to get this functionality.

Reply:
Having checked, it looks like it does duck out to the database

```
   (0.8ms)  SELECT "posts"."id" FROM "posts" WHERE ("posts"."ancestry" LIKE '178519/%' OR "posts"."ancestry" = '178519')
  ↳ app/views/posts/_post_subject.html.erb:12
```

Changing the code to
```ruby
post.descendants.count(:all)
```

gives the alternate query
```
   (0.3ms)  SELECT COUNT(*) FROM "posts" WHERE ("posts"."ancestry" LIKE '178519/%' OR "posts"."ancestry" = '178519')
  ↳ app/views/posts/_post_subject.html.erb:1
```

Reply:
Huh, neat! I'll admit I just gave a quick glance at the logs and Ancestry library last night, still learning how to parse Ruby.